src/addrman.cpp

```diff
@@ -805,11 +805,17 @@ int AddrManImpl::GetEntry(bool use_tried, size_t bucket, size_t position) const
     return -1;
 }
 
-std::vector<CAddress> AddrManImpl::GetAddr_(size_t max_addresses, size_t max_pct, std::optional<Network> network, const bool filtered) const
+std::vector<CAddress> AddrManImpl::GetAddr_(size_t max_addresses, size_t max_pct, const std::optional<Network>& network, const bool filtered) const
 {
     AssertLockHeld(cs);
 
     size_t nNodes = vRandom.size();
+    if (network) {
+        auto it = m_network_counts.find(*network);
+        if (it == m_network_counts.end()) return {};
+        const auto counts{it->second};
+        nNodes = counts.n_new + counts.n_tried;
+    }
```

Comment:
nit: To make it clear that we're overwriting the same value:
```suggestion
    size_t nNodes;
    if (network) {
        auto it = m_network_counts.find(*network);
        if (it == m_network_counts.end()) return {};
        nNodes = it->second.n_new + it->second.n_tried;
    } else {
        nNodes = vRandom.size();
    }
```

Reply:
I'll leave it as is for now.